src/sentry/preprod/models.py

```diff
@@ -0,0 +1,222 @@
+        """The upload is complete, but the artifact is not yet processed."""
+        PROCESSED = 3
+        """The artifact has been processed and is ready to be used."""
+        FAILED = 4
+        """The artifact failed to upload or process. Read the error_code and error_message for more details."""
+
+        @classmethod
+        def as_choices(cls):
+            return (
+                (cls.UPLOADING, "uploading"),
+                (cls.UPLOADED, "uploaded"),
+                (cls.PROCESSED, "processed"),
+                (cls.FAILED, "failed"),
+            )
+
+    class ArtifactType(IntEnum):
+        XCARCHIVE = 0
+        """Apple Xcode archive."""
+        AAB = 1
+        """Android App Bundle."""
+        APK = 2
+        """Android APK."""
+
+        @classmethod
+        def as_choices(cls):
+            return (
+                (cls.XCARCHIVE, "xcarchive"),
+                (cls.AAB, "aab"),
+                (cls.APK, "apk"),
+            )
+
+    class ErrorCode(IntEnum):
+        UNKNOWN = 0
+        """The error code is unknown. Try to use a descriptive error code if possible."""
+        UPLOAD_TIMEOUT = 1
+        """The upload timed out."""
+        ARTIFACT_PROCESSING_TIMEOUT = 2
+        """The artifact processing timed out."""
+        ARTIFACT_PROCESSING_ERROR = 3
+        """The artifact processing failed."""
+
+        @classmethod
+        def as_choices(cls):
+            return (
+                (cls.UNKNOWN, "unknown"),
+                (cls.UPLOAD_TIMEOUT, "upload_timeout"),
+                (cls.ARTIFACT_PROCESSING_TIMEOUT, "artifact_processing_timeout"),
+                (cls.ARTIFACT_PROCESSING_ERROR, "artifact_processing_error"),
+            )
+
+    __relocation_scope__ = RelocationScope.Excluded
+
+    project = FlexibleForeignKey("sentry.Project")
+
+    # Nullable in case the file upload fails
+    file = FlexibleForeignKey("sentry.File", null=True)
+
+    # The date the artifact was built. E.g. an artifact could be built on 05/21/2025,
+    # but the user uploaded it on 05/22/2025.
+    date_built = models.DateTimeField(null=True)
+
+    build_configuration = FlexibleForeignKey(
+        "preprod.PreprodBuildConfiguration", null=True, on_delete=models.SET_NULL
+    )
+
+    state = BoundedPositiveIntegerField(
+        default=ArtifactState.UPLOADING, choices=ArtifactState.as_choices()
+    )
+
+    # Nullable because we only know the type after the artifact has been processed
+    artifact_type = BoundedPositiveIntegerField(choices=ArtifactType.as_choices(), null=True)
+
+    error_code = BoundedPositiveIntegerField(choices=ErrorCode.as_choices(), null=True)
+    error_message = models.TextField(null=True)
+
+    # E.g. 1.2.300
+    build_version = models.CharField(max_length=255, null=True)
+    # E.g. 9999
+    build_number = BoundedBigIntegerField(null=True)
+
+    extras: models.Field[dict[str, Any], dict[str, Any]] = JSONField(null=True)
+
+    class Meta:
+        app_label = "preprod"
```

Comment:
You'll need to add `preprod` here now as well:
https://github.com/getsentry/sentry/blob/e165efbf3c7bf45ae26c46eb0a8fbea81cce0692/src/sentry/conf/server.py#L3193-L3207